src/BccCode.Linq/ApiClient/ResultList.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System.Collections;
+using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace BccCode.ApiClient;
+
+[DataContract(Name = "ResultList")]
+public abstract class ResultList
+{
+    protected ResultList(Metadata? meta = null)
+    {
+    }
+
+    /// <summary>
+    /// Gets or Sets Meta
+    /// </summary>
+    [DataMember(Name = "meta", EmitDefaultValue = false)]
+    public Metadata? Meta { get; set; }
+
+    [JsonIgnore]
+    public IEnumerable<object>? Data { get; set; }
+
+    public abstract void AddData(IEnumerable? data);
+
+    [Obsolete($"Use property {nameof(Data)} instead")]
+    public virtual IEnumerable<object> GetData() => Data ?? new List<object>();
+
+    [Obsolete($"Use property {nameof(Data)}.Count() instead")]
+    public virtual long GetCount() => Data?.LongCount() ?? 0;
+}
+
+
+public class ResultList<T> : ResultList, IResultList<T>, IEquatable<ResultList<T>>, IValidatableObject
+{
+    public ResultList()
+    {
+    }
+
+    public ResultList(ResultList<T> resultList)
+    {
+        Data = resultList.Data;
+        Meta = resultList.Meta;
+    }
+
+    public ResultList(IResultList<T> resultList)
```

Comment:
Do we still need IResultList<>?

Reply:
For sure. It wasn't my invention. I can do an update to drop it.